components/Admin/Admin.js

```diff
@@ -22,35 +24,67 @@ class Admin extends React.Component {
 
   handleSearch = (event) => {
     const searchTerm = event.currentTarget.value
-    let { locations, filteredLocations } = this.state
+    let { locations, matchingSearchLocations } = this.state
 
     if (searchTerm) {
-      filteredLocations = locations.filter(location =>
+      matchingSearchLocations = locations.filter(location =>
         location.name.indexOf(searchTerm) >= 0)
     } else {
-      filteredLocations = null
+      matchingSearchLocations = null
     }
 
-    this.setState({
-      locations,
-      filteredLocations,
-    })
+    this.setState({ matchingSearchLocations })
   }
 
   handleNewFacility = () => {
 
   }
 
-  handleCategoryFilter = (category) => {
+  handleCategoryFilter = (category, active=false) => {
+    let { locations, selectedCategories, matchingCategoryLocations } = this.state
+
+    if (active) {
+      selectedCategories.push(category.taxonomy)
```

Comment:
Same deal as above comment. I am (maybe naively) worried about doing a mutation without `setState` that could cause race conditions.

And instead of `push`, we can do a non-mutative append by:

```const newSelectedCategories = [...selectedCategories, category.taxonomy]
```

Some of these transforms in this file might be much better w/ either lodash or ramda or immutable, but let's see how annoying the JS becomes first